src/main/kotlin/org/eclipse/uprotocol/rpc/RpcMapper.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.uprotocol.rpc
+
+import com.google.protobuf.Any
+import com.google.protobuf.InvalidProtocolBufferException
+import com.google.protobuf.Message
+import com.google.rpc.Code
+import com.google.rpc.Status
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionException
+import org.eclipse.uprotocol.transport.datamodel.UPayload
+
+/**
+ * RPC Wrapper is an interface that provides static methods to be able to wrap an RPC request with
+ * an RPC Response (uP-L2). APIs that return Message assumes that the payload is protobuf serialized
+ * com.google.protobuf.Any (UPayloadFormat.PROTOBUF) and will barf if anything else is passed
+ */
+interface RpcMapper {
+    companion object {
+        /**
+         * Map a response of CompletableFuture&lt;UPayload&gt; from Link into a CompletableFuture containing the declared expected return type of the RPC method or an exception.
+         * @param responseFuture CompletableFuture&lt;UPayload&gt; response from uTransport.
+         * @param expectedClazz The class name of the declared expected return type of the RPC method.
+         * @return Returns a CompletableFuture containing the declared expected return type of the RPC method or an exception.
+         * @param <T> The declared expected return type of the RPC method.
+        </T> */
+        fun <T : Message> mapResponse(
+            responseFuture: CompletableFuture<UPayload>,
```

Comment:
should we consider coroutines here (i.e. kotlin flow or suspendable functions)? CompletableFuture is the java async type.